code/modules/overmap/exoplanets/turfs.dm

```diff
@@ -76,200 +86,200 @@
 	icon_state = "greygrass"
 	color = "#799c4b"
 	footstep_type = /decl/footsteps/grass
 
-/turf/simulated/floor/exoplanet/grass/Initialize()
+/turf/exterior/grass/Initialize()
 	. = ..()
-	if(GLOB.using_map.use_overmap)
-		var/obj/effect/overmap/visitable/sector/exoplanet/E = map_sectors["[z]"]
-		if(istype(E) && E.grass_color)
-			color = E.grass_color
-	if(!resources)
-		resources = list()
+	if(owner && owner.grass_color)
+		color = owner.grass_color
+
+	var/datum/extension/buried_resources/resources = get_or_create_extension(src, /datum/extension/buried_resources)
 	if(prob(70))
-		resources[/decl/material/solid/mineral/graphite] = rand(3,5)
+		LAZYSET(resources.resources, /decl/material/solid/mineral/graphite, rand(3,5))
 	if(prob(5))
-		resources[/decl/material/solid/metal/uranium] = rand(1,3)
+		LAZYSET(resources.resources, /decl/material/solid/metal/uranium, rand(1,3))
 	if(prob(2))
-		resources[/decl/material/solid/gemstone/diamond] = 1
+		LAZYSET(resources.resources, /decl/material/solid/gemstone/diamond, 1)
+	if(!length(resources.resources))
+		remove_extension(src, /datum/extension/buried_resources)
 
-/turf/simulated/floor/exoplanet/grass/fire_act(datum/gas_mixture/air, exposed_temperature, exposed_volume)
+/turf/exterior/grass/fire_act(datum/gas_mixture/air, exposed_temperature, exposed_volume)
 	if((temperature > T0C + 200 && prob(5)) || temperature > T0C + 1000)
 		melt()
 
-/turf/simulated/floor/exoplanet/grass/melt()
+/turf/exterior/grass/melt()
 	SetName("scorched ground")
 	icon_state = "scorched"
 	footstep_type = /decl/footsteps/asteroid
 	color = null
 
 //Sand
-/turf/simulated/floor/exoplanet/desert
+/turf/exterior/desert
 	name = "sand"
 	desc = "It's coarse and gets everywhere."
 	dirt_color = "#ae9e66"
 	footstep_type = /decl/footsteps/sand
 
-/turf/simulated/floor/exoplanet/desert/Initialize()
+/turf/exterior/desert/Initialize()
 	. = ..()
 	icon_state = "desert[rand(0,5)]"
 
-/turf/simulated/floor/exoplanet/desert/fire_act(datum/gas_mixture/air, exposed_temperature, exposed_volume)
+/turf/exterior/desert/fire_act(datum/gas_mixture/air, exposed_temperature, exposed_volume)
 	if((temperature > T0C + 1700 && prob(5)) || temperature > T0C + 3000)
 		melt()
 
-/turf/simulated/floor/exoplanet/desert/melt()
+/turf/exterior/desert/melt()
 	SetName("molten silica")
 	desc = "A glassed patch of sand."
 	icon_state = "sandglass"
 	diggable = 0
 
 //Concrete
-/turf/simulated/floor/exoplanet/concrete
+/turf/exterior/concrete
 	name = "concrete"
 	desc = "Stone-like artificial material."
 	icon = 'icons/turf/flooring/misc.dmi'
 	icon_state = "concrete"
 
-/turf/simulated/floor/exoplanet/concrete/on_update_icon()
-	cut_overlays()
-	if(burnt)
-		add_overlay(get_damage_overlay("burned[(x + y) % 3]", BLEND_MULTIPLY))
-	if(broken)
-		add_overlay(get_damage_overlay("broken[(x + y) % 5]", BLEND_MULTIPLY))
-
-/turf/simulated/floor/exoplanet/concrete/melt()
-	burnt = TRUE
-	update_icon()
+/turf/exterior/concrete/on_update_icon()
+	return
```

Comment:
what about bork overlays?
Also wouldn't this make gas overlays not show up on concrete?